lib/winapi/winnt.c

```diff
@@ -0,0 +1,26 @@
+#include <winnt.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+LONG64 InterlockedExchange64 (LONG64 volatile *Target, LONG64 Value)
+{
+    // There's no xchg8b in x86, so we implement it with cmpxchg8b
+    LARGE_INTEGER li;
+    li.QuadPart = *Target;
+    __asm__ __volatile__("1:\n"
+                         "cmpxchg8b (%2)\n"
+                         "jnz 1b\n" // If the comparison failed, try again with updated edx:eax
+                         : "+a"(li.LowPart), "+d"(li.HighPart)
+                         : "r"(Target), "b"((int)(Value & 0xFFFFFFFF)), "c"((int)(Value >> 32))
```

Comment:
Why are you using `LARGE_INTEGER` for `Target` in `eax:edx` but not for `Value`?

Reply:
No particular reason. I may end up changing `Value`.

Reply:
I modified the code to use `LARGE_INTEGER` for the value, too, and slightly tweaked my comments.